### PR TITLE
[DRUP-796] Replace apigee_edge_apidocs with apigee_api_catalog

### PR DESCRIPTION
**Pull request by shadcn**
This PR replaces the apigee_edge_apidcos with the new apigee_api_catalog module.

1. I've updated the configuration and dependencies.
2. Added the requirements in composer.json

Note: this pulls in the `8.x-1.0-alpha1` release for which does not have the path rename yet (apidoc to apis). 

We can create a follow up PR to add this later.

### CODE REVIEW

File: config/install/views.view.apigee_api_catalog.yml [hunk 1]
```diff
@@ -3,19 +3,18 @@ status: true
 dependencies:
   config:
     - core.entity_view_mode.apidoc.card
-    - field.storage.apidoc.field_categories
```
Comment (jacine) [inline config/install/views.view.apigee_api_catalog.yml:6]: I'm not sure if this should be removed.... Going to try to find out.

Comment (jacine) [reply config/install/views.view.apigee_api_catalog.yml:6]: I asked today, and it seems this field should remain.

**Review by jacine — CHANGES_REQUESTED** [review]
Marking this as needed changes until we get the Categories field back.

**Comment by shadcn** [comment]
This is on hold pending https://github.com/apigee/apigee-api-catalog-drupal/pull/7

File: config/install/views.view.apigee_api_catalog.yml (hunk 1 above)
Comment (jacine) [reply config/install/views.view.apigee_api_catalog.yml:6]: @arshad and I figured out this was left over from older `apigee_edge_apidocs` code, so this was a false alarm.  It's all good.

**Review by jacine — APPROVED** [review]
I've tested this, and it's all good!  Thanks @arshad! 👏

**Comment by shadcn** [comment]
https://github.com/apigee/apigee-api-catalog-drupal/pull/7 was merged and @cnovak released https://github.com/apigee/apigee-api-catalog-drupal/releases/tag/8.x-1.0-beta1.

@jacine This is ready for another quick review and merge.

File: composer.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "drupal-composer/drupal-scaffold": "^2",
     "drupal/admin_toolbar": "^1.0",
     "drupal/adminimal_admin_toolbar": "^1.9",
-    "drupal/apigee_api_catalog": "~1.0",
+    "drupal/apigee_api_catalog": "^1.0@beta",
```
Comment (cnovak) [inline composer.json]: Why change this? The ~1.0 will pull in latest instead of being tied to beta versions

**Review by cnovak — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (shadcn) [reply composer.json]: Fixed this. Thanks.

**Review by jacine — APPROVED** [review]

**Review by cnovak — APPROVED** [review]

**Pull request merged**